shared/test/fake_redis_client.rb

```diff
@@ -29,8 +29,15 @@ def del(key)
   end
 
   def hdel(key, field)
-    value = get_hash_for_key(key).delete(field)
-    value ? 1 : 0
+    if field.is_a?(Array)
```

Comment:
we can hoist "hash = get_hash_for_key(key)" up here to avoid repeated calls in the loop.